### DISCUSSION TITLE
CVSS-Tampering: GitHub creates incorrect CVSS v4 scores

### DISCUSSION THREAD

**Discussion by AB-xdev** [aquasecurity/trivy#8042] [False Detection]
### IDs

Probably most recent CVEs in GitHub advisories database

#### The problem

GitHub ([Advisories](https://github.com/github/advisory-database)) seems to create CVSS v4 scores for *some* advisories as I found out in https://github.com/github/advisory-database/pull/5032 / https://github.com/github/advisory-database/issues/5058.

There are some problem with that, here is a quick recap:
* This is only done for certain CVEs, not all
* The process how these CVSS v4 values are created is not transparent
  * There is no (public) documentation that this is done, how this is done and why it is done
  * Likely no communication with CVE creators / CVSSv4 values do not match CVE descriptions (usually only contains CVSS v3 score explanation)
* Created CVSS v4 are not marked as "computed from CVSSv3 by GitHub" or something similar anywhere
* Original CVSS score is not used for computing the ``severity``

This process seems to result in incorrect scores (some values do not match at all) and incorrect ``severity`` values, thus resulting in False Postives and Negatives.

Spontaneously found examples:
| CVE | CVSS v3 (original) | CVSS v4 created by GitHub (used in ``severity``)  | Note |
|  --- | --- | --- | --- |
| CVE-2024-47535 | [5.5 moderate](https://github.com/netty/netty/security/advisories/GHSA-xq3w-v528-46rv) | [7.0 High](https://github.com/advisories/GHSA-xq3w-v528-46rv) | Incorrect metrics: ``SC``, ``SI`` |
| CVE-2024-53848 | [7.1 High](https://github.com/python-jsonschema/check-jsonschema/security/advisories/GHSA-q6mv-284r-mp36) | [6.1 Moderate](https://github.com/advisories/GHSA-q6mv-284r-mp36) | ``Vulnerable System Impact Metrics`` seem to be missing |
| CVE-2024-52806 | [8.3 High](https://github.com/simplesamlphp/saml2/security/advisories/GHSA-pxm4-r5ph-q2m2) | [6.9 Medium](https://github.com/advisories/GHSA-pxm4-r5ph-q2m2) | ``Vulnerable System Impact Metrics`` seem to be missing |

Further asked questions by me have been evaded/dismissed so far.

The persons responsible for creating CVSS v4 scores also seem to have some trouble in doing so as their proposed values have been incorrect now for 2 times despite having clear points on what specific metrics are wrong.

#### Recommended actions

GitHub Advisories seems to see 0 problems with this behavior as can be seen in the comments, so I don't expect them changing their opinion any time soon without further external aspects.

I would therefore propose that you exclude the GitHub's advisory database for now or ignore ``CVSSv4`` and ``severity`` and only use trustworthy alternatives like e.g. [GitLab's database](https://gitlab.com/gitlab-org/security-products/gemnasium-db) instead.

### Scanner

Vulnerability

### Checklist

- [X] Read [the documentation regarding wrong detection](https://aquasecurity.github.io/trivy/dev/community/contribute/discussion/#false-detection)
- [X] Ran Trivy with `-f json` that shows data sources and confirmed that the security advisory in data sources was correct

**Comment by knqyf263** [comment]
I read through https://github.com/github/advisory-database/pull/5032, and your comments make sense to me. However, other advisories also sometimes score unreasonable values. I think we have to discuss with the issuer for each advisory, although it's not ideal. GitHub might change their mind if many users complain about it.

Since removing the GHSA would have a major impact, it might be a good idea to provide an option to select the severity source, like `--severity-src gitlab` and allow each user to choose.

**Reply by AB-xdev** [reply]
> Since removing the GHSA would have a major impact, it might be a good idea to provide an option to select the severity source, like ``--severity-src gitlab`` and allow each user to choose.

Very good idea +1
